tests/unit/amo/components/TestUserProfileEdit.js

```diff
@@ -0,0 +1,601 @@
+    const location = 'new location';
+    root.find(`.UserProfileEdit-location`).simulate(
+      'change',
+      createFakeEventChange({
+        name: 'location',
+        value: location,
+      })
+    );
+    root.find('.UserProfileEdit-form').simulate('submit', createFakeEvent());
+
+    sinon.assert.calledWith(dispatchSpy, editUserAccount({
+      errorHandlerId: errorHandler.id,
+      userFields: {
+        biography: user.biography,
+        display_name: user.displayName,
+        homepage: user.homepage,
+        location,
+        occupation: user.occupation,
+        username: user.username,
+      },
+      userId: user.id,
+    }));
+  });
+
+  it('renders a success message when user profile has been updated', () => {
+    const { store } = signInUserWithUsername('tofumatt');
+    const user = getCurrentUser(store.getState().users);
+
+    const occupation = 'new occupation';
+
+    _editUserAccount({
+      store,
+      userFields: {
+        occupation,
+      },
+      userId: user.id,
+    });
+
+    const root = renderUserProfileEdit({ store });
+
+    expect(root.find(Notice)).toHaveLength(0);
+    expect(root.find('.UserProfileEdit-submit-button'))
+      .toHaveProp('disabled', true);
+
+    // The user profile has been updated.
+    store.dispatch(finishEditUserAccount());
+
+    const { isEditing } = store.getState().users;
+    root.setProps({ isEditing });
+
+    expect(root.find(Notice)).toHaveLength(1);
+    expect(root.find(Notice)).toHaveProp('type', 'success');
+    expect(root.find(Notice))
+      .toHaveProp('children', 'Profile successfully updated.');
+
+    expect(root.find('.UserProfileEdit-submit-button'))
+      .toHaveProp('disabled', false);
+  });
+
+  it('does not render a success message when an error occured', () => {
+    const { store } = signInUserWithUsername('tofumatt');
+    const user = getCurrentUser(store.getState().users);
+
+    const errorHandler = new ErrorHandler({
+      id: 'some-id',
+      dispatch: store.dispatch,
+    });
+
+    const username = '';
+
+    _editUserAccount({
+      errorHandlerId: errorHandler.id,
+      store,
+      userFields: {
+        username,
+      },
+      userId: user.id,
+    });
+
+    const root = renderUserProfileEdit({ errorHandler, store });
+
+    expect(root.find(Notice)).toHaveLength(0);
+    expect(root.find('.UserProfileEdit-submit-button'))
+      .toHaveProp('disabled', true);
+
+    // An error occured while updating the user profile.
+    errorHandler.handle(new Error('unexpected error'));
+    store.dispatch(finishEditUserAccount());
+
+    const { isEditing } = store.getState().users;
+    root.setProps({ isEditing });
+
+    expect(root.find(Notice)).toHaveLength(0);
+
+    expect(root.find('.UserProfileEdit-submit-button'))
+      .toHaveProp('disabled', false);
+  });
+
+  it('renders a Not Found page when logged user cannot edit another user', () => {
+    const username = 'current-logged-user';
```

Comment:
Nit: `current-logged-in-user`.

Reply:
I'm going to stop commenting on each of these, but I think any instance of "logged user" should say "logged-in user" (or "logged in user").

Reply:
yes, I am so sorry for that :/